### PR TITLE
[elixir] fix: athena error message when partition exists

### PR DESCRIPTION
**Pull request by grejdi-mbta**

### CODE REVIEW

File: ex_cubic_ingestion/lib/ex_aws/helpers.ex [hunk 1]
```diff
@@ -108,7 +108,7 @@ defmodule ExAws.Helpers do
   defp query_succeeded?(%{
          "Status" => %{
            "State" => "FAILED",
-           "AthenaError" => %{"ErrorMessage" => "Partition entries already exist."}
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_aws/helpers.ex:111]: Not sure if I had a typo, but dev is returning different message now.

File: ex_cubic_ingestion/lib/ex_aws/helpers.ex [hunk 2]
```diff
@@ -108,7 +108,7 @@ defmodule ExAws.Helpers do
...
+           "AthenaError" => %{"ErrorMessage" => "Partition already exist."}
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_aws/helpers.ex:111]: Was hoping to use an error code, but Athena is surfacing this from Hadoop. Their error codes are rather shallow: https://docs.aws.amazon.com/athena/latest/ug/error-reference.html

File: ex_cubic_ingestion/lib/ex_aws/helpers.ex (hunk 1 above)
Comment (paulswartz) [reply ex_cubic_ingestion/lib/ex_aws/helpers.ex:111]: I'm still seeing "Partition entries already exist." in dev?
```
{\\\"AthenaError\\\":{\\\"ErrorCategory\\\":2,\\\"ErrorMessage\\\":\\\"Partition entries already exist.\\\",\\\"ErrorType\\\":1006,\\\"Retryable\\\":false}
```

File: ex_cubic_ingestion/test/ex_aws/helpers_test.exs [hunk 3]
```diff
@@ -94,8 +94,17 @@ defmodule ExAws.HelpersTest do
         {:ok, %{"QueryExecutionId" => "already_added_partition_query_id"}}
       ]
 
-      assert {:error, _message} =
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/test/ex_aws/helpers_test.exs:97]: This test was wrong in its assertion.

File: ex_cubic_ingestion/test/ex_aws/helpers_test.exs [hunk 4]
```diff
@@ -94,8 +94,17 @@ defmodule ExAws.HelpersTest do
...
-               ExAws.Helpers.monitor_athena_query_executions(MockExAws, query_executions)
+      assert :ok = ExAws.Helpers.monitor_athena_query_executions(MockExAws, query_executions)
+    end
+
+    # note: multiple added partitions
+    test "returns success even if we have a failed query due to already added partitions" do
+      query_executions = [
+        {:ok, %{"QueryExecutionId" => "success_query_id"}},
+        {:ok, %{"QueryExecutionId" => "already_added_partitions_query_id"}}
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/test/ex_aws/helpers_test.exs:104]: This occurs when multiple partitions are added, i.e. snapshot and identifier for ODS loads.

File: ex_cubic_ingestion/lib/ex_aws/helpers.ex (hunk 1 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_aws/helpers.ex]: Fixed now. I need to look at it more closely. If it's one partition, such as DMAP loads with just 'identifier', it throws the `Partition already exists.` message. If it's multiple partitions, such as ODS loads with 'snapshot' and 'identifier', it throws the `Partition entries already exist.`

File: ex_cubic_ingestion/test/support/ex_aws.ex [hunk 5]
```diff
@@ -423,21 +423,23 @@ defmodule MockExAws do
   def request(
         %{
           service: :athena,
-          data: %{QueryString: "MSCK REPAIR TABLE success_table;"}
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/test/support/ex_aws.ex:426]: Removed these older test responses.

**Review by paulswartz — APPROVED** [review]

**Pull request merged**